### ISSUE TITLE
ChefDK fails with 404

### ISSUE THREAD

**Issue by luisdavim** [caskroom/homebrew-cask#39293]
#### General troubleshooting steps

- [x] I have checked the instructions for [reporting bugs](https://github.com/caskroom/homebrew-cask#reporting-bugs) (or [making requests](https://github.com/caskroom/homebrew-cask#requests)) before opening the issue.
  - [x] None of the templates was appropriate for my issue, or I’m not sure.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/caskroom/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

#### Description of issue

when trying to install chefDK the process fails with an error

#### Output of your command with `--verbose --debug`

```
curl: (22) The requested URL returned error: 404
Error: Download failed on Cask 'chefdk' with message: Download failed: https://packages.chef.io/files/stable/chefdk/2.3.4/mac_os_x/10.13/chefdk-2.3.4-1.dmg
```

**Comment by vitorgalvao** [comment]
> [If you ignore this guide, your issue may be closed without review](https://github.com/caskroom/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

Also note from that document:

> If the maintainer was wrong in closing your issue, please do reply stating why! Closing an issue does not mean the conversation is over. If the guides themselves were unclear, help us improve them! Open *first* an issue or pull request stating what you found confusing *and only then* your other issue.

---

https://github.com/caskroom/homebrew-cask/blob/master/doc/reporting_bugs/a_cask_fails_to_install.md#curl-error

**Comment by luisdavim** [comment]
Hi, this issue should not be closed, cask is trying using the url:
`https://packages.chef.io/files/stable/chefdk/2.3.4/mac_os_x/10.13/chefdk-2.3.4-1.dmg`
but that doesn't exist, the latest version available on chef's website is:
`https://packages.chef.io/files/stable/chefdk/2.3.4/mac_os_x/10.12/chefdk-2.3.4-1.dmg`

see here: https://downloads.chef.io/chefdk

I think the problem is that I'm on macOS High-Sierra (10.13) and the latest package available is for macOS Sierra (10.12).

**Comment by vitorgalvao** [comment]
https://github.com/caskroom/homebrew-cask/pull/39319
